mindmeld/components/entity_resolver.py

```diff
@@ -42,60 +46,284 @@
     load_index,
     resolve_es_config_for_version,
 )
+from .. import path
+from ..core import Entity
+from ..exceptions import EntityResolverConnectionError, EntityResolverError
 
 logger = logging.getLogger(__name__)
 
 
+def _is_module_available(module_name: str):
+    """
+    checks if a module is available or not (eg. _is_module_available("sentence_transformers"))
+
+    Args:
+        module_name (str): name of the model to check
+
+    Returns:
+        bool, if or not the given module exists
+    """
+    return bool(importlib.util.find_spec(module_name) is not None)
+
+
+def _get_module_or_attr(module_name: str, func_name: str = None):
+    """
+    Loads an attribute from a module or a module itself
+    (check if the module exists before calling this function)
+    """
+    m = importlib.import_module(module_name)
+    if not func_name:
+        return m
+    if func_name not in dir(m):
+        raise ImportError(f"Cannot import {func_name} from {module_name}")
+    return getattr(m, func_name)
+
+
+SBERT_AVAILABLE = _is_module_available("sentence_transformers")
+if SBERT_AVAILABLE:
```

Comment:
Can these also be loaded dynamically, that is, at `if name == "sbert_cosine_similarity" and not SBERT_AVAILABLE:`, can we dynamic load if its not available? I would like us to avoid these condition specific import executing at the top level of the script.

Reply:
Removed loading them at the top level.